js/search/search.js

```diff
@@ -519,160 +517,33 @@ class Search {
         }
     }
 
-}
-
-/**
- * Deleting the data index folder
- * when the app is closed/signed-out/navigates
- */
-function deleteIndexFolder() {
-    Search.deleteIndexFolders(searchConfig.FOLDERS_CONSTANTS.INDEX_PATH);
-}
-
-/**
- * Reads the file from the msgjson
- * this is only for the demo page
- * @param batch
- * @returns {Promise<Array>}
- */
-function readFile(batch) {
-    return new Promise((resolve, reject) => {
-        let dirPath = path.join(searchConfig.FOLDERS_CONSTANTS.EXEC_PATH, isMac ? '..' : '', 'msgsjson', batch);
-        let messageFolderPath = isDevEnv ? path.join('./msgsjson', batch) : dirPath;
-        let files = fs.readdirSync(messageFolderPath);
-        this.messageData = [];
-        files.forEach((file) => {
-            let tempPath = path.join(messageFolderPath, file);
-            let data = fs.readFileSync(tempPath, "utf8");
-            if (data) {
-                try {
-                    this.messageData.push(JSON.parse(data));
-                } catch (err) {
-                    reject(new Error(err))
-                }
-            } else {
-                reject(new Error('Error reading batch'))
-            }
-        });
-        resolve(this.messageData);
-    });
-}
-
-/**
- * Creating launch agent for handling the deletion of
- * index data folder when app crashed or on boot up
- */
-function initializeLaunchAgent() {
-    let pidValue = process.pid;
-    if (isMac) {
-        createLaunchScript(pidValue, 'clear-data', searchConfig.LIBRARY_CONSTANTS.LAUNCH_AGENT_FILE, function (res) {
-            if (!res) {
-                log.send(logLevels.ERROR, `Launch Agent not created`);
-            }
-            createLaunchScript(null, 'clear-data-boot', searchConfig.LIBRARY_CONSTANTS.LAUNCH_DAEMON_FILE, function (result) {
-                if (!result) {
-                    log.send(logLevels.ERROR, `Launch Agent not created`);
-                }
-                launchDaemon(`${searchConfig.FOLDERS_CONSTANTS.USER_DATA_PATH}/.symphony/clear-data-boot.sh`, function (data) {
-                    if (data) {
-                        log.send(logLevels.INFO, 'Launch Daemon: Creating successful');
-                    }
-                });
-            });
-
-            launchAgent(pidValue, `${searchConfig.FOLDERS_CONSTANTS.USER_DATA_PATH}/.symphony/clear-data.sh`, function (response) {
-                if (response) {
-                    log.send(logLevels.INFO, 'Launch Agent: Creating successful');
-                }
-            });
-        });
-    } else {
-        let folderPath = isDevEnv ? path.join(__dirname, '..', '..', searchConfig.FOLDERS_CONSTANTS.INDEX_FOLDER_NAME) :
-            path.join(searchConfig.FOLDERS_CONSTANTS.USER_DATA_PATH, searchConfig.FOLDERS_CONSTANTS.INDEX_FOLDER_NAME);
-        taskScheduler(`${searchConfig.LIBRARY_CONSTANTS.WINDOWS_TASK_FILE}`, folderPath, pidValue, `${searchConfig.LIBRARY_CONSTANTS.WINDOWS_CLEAR_SCRIPT}`);
+    if (this.userId) {
```

Comment:
Another potential place of exploitation. We need to add more validations if possible.

Reply:
@keerthi16 Once this is addressed, I'll approve this PR.

Reply:
I'm not really sure how we need to handle this as this is just a random number, and this is just used to name the index file.